opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/RevealJsonFormFragmentPresenter.java

```diff
@@ -61,9 +64,16 @@ public ValidationStatus writeValuesAndValidate(LinearLayout mainView) {
                             onLocationValidated();
                         }
                     }
+                    break;
                 }
             }
         }
+        if (!hasMapView && validationStatus.isValid()) {
+            onLocationValidated();
+        } else {
+            Toast.makeText(getView().getContext(), validationStatus.getErrorMessage(), Toast.LENGTH_LONG).show();
```

Comment:
Won't this block run when there is a mapview in the form?

Should only run if `!validationStatus.isValid()`?

Reply:
Yes  when there is a mapview and mapview does not have any validations errors then 
 `validateUserLocation(mapView) `or  `onLocationValidated()` will be called. These will be if user location validation is required or if not then trigger save.

This block will run only if
- There was no map on form and form is valid
- There was no map on form but form is invalid
- there was map on form but the form is invalid or map is invalid.

Reply:
If you look at the `else` block, it runs when there is every time there is no mapview, even if the form is valid, right?

`* there was map on form but the form is invalid or map is invalid.`

I don't think the statement above is correct.

Reply:
Yes the else is for the these last conditions

> * There was no map on form but form is invalid
> * there was map on form but the form is invalid or map is invalid.